### PR TITLE
Fix bug #1367713 - JMX/Threads console is blank

### PR DESCRIPTION
**Pull request by astefanutti**

### CODE REVIEW

**Comment by spadgett** [comment]
@astefanutti The index.html segment was removed to fix another bug. See

https://github.com/openshift/origin/commit/8c3be50fbd7df6c5410b51b19772826ffeb8b822

Have you been able to build and verify this fix?

**Comment by astefanutti** [comment]
@spadgett I've tested it on Chrome, didn't know about the previous Safari issue. Let me test it on Safari as well...

**Comment by astefanutti** [comment]
@spadgett the thing is that with openshift/origin@8c3be50 the trailing slash is removed which causes the error.

**Comment by spadgett** [comment]
I think a URI.js update might be normalizing the URL to remove the trailing slash.

**Comment by spadgett** [comment]
@astefanutti This might work?

```js
            var targetURI = new URI().path(BaseHref)
                                     .segment('java')
                                     // Add an empty segment to force a trailing slash, which is required.
                                     .segment('')
                                     .hash(token)
                                     .query({
                                       jolokiaUrl: jolokiaUrl,
                                       title: title,
                                       returnTo: returnTo
                                     });
```

**Comment by astefanutti** [comment]
@spadgett thanks. Let me try that out...

**Comment by spadgett** [comment]
@astefanutti Can you verify that the font icons load properly as well? They aren't loading for me.

**Comment by astefanutti** [comment]
@spadgett adding an empty segment seems to work. I've just repushed with it.

On my local environment, font icons load correctly on Chrome but not on Safari.

**Comment by jwforres** [comment]
@astefanutti even in chrome we see it trying to load the font icon from ../components/ path which doesn't exist

File: app/scripts/extensions/javalink/javaLink.js [hunk 1]
```diff
@@ -66,7 +66,9 @@
             var title = container.name || 'Untitled Container';
             var token = AuthService.UserStore().getToken() || '';
             var targetURI = new URI().path(BaseHref)
-                                     .segment('java/') // Must have a trailing slash to avoid runtime errors in Safari
+                                     .segment('java')
+                                     // Add an empty segment to force a trailing slash, which is required.
+                                     .segment('/')
```
Comment (spadgett) [inline app/scripts/extensions/javalink/javaLink.js]: I believe this should be

```js
.segment('')
```

See https://github.com/medialize/URI.js/commit/2b604927f997299789c50badadbbb295c3bab754

**Comment by jwforres** [comment]
@astefanutti are you running it embedded in the openshift binary? because we get a release of openshift-jvm with everything packaged into a zip and thats what gets embedded into openshift, and in that setup the fonts are in a /fonts directory

**Comment by astefanutti** [comment]
@spadgett my mistake (still works though). I've just repushed.

The original regression may have been introduced with  medialize/URI.js@0f8e416ee649cda2c76d34fd26e7702beb82258a.

**Comment by astefanutti** [comment]
@jwforres no, I'm running it locally.

**Comment by jwforres** [comment]
This is what gets released to us from you guys:

```
curl -s https://codeload.github.com/hawtio/openshift-jvm/tar.gz/v1.0.45-build | tar -xz -C openshift-jvm --strip-components=1
```

And in that zip if you look at what the style.css is referencing for the fontawesome font, its pointing to a path that doesn't exist.  Looks like it has the same problem with the glyphicons for bootstrap.

**Comment by jwforres** [comment]
@astefanutti so looking back in the history we can see there use to be two copies of the fontawesome CSS, one had the correct paths and one had the incorrect paths, between v1.0.44 and v1.0.45 the duplicate copy was taken out, but the one that was taken out was the one with the correct paths.  I'm going to open a separate PR that rolls openshift back to 1.0.42 which is what we were at before @gashcrumb bumped the version last week.  Unless there is a strong reason for us to update to a later version?

**Comment by jwforres** [comment]
[merge] for this fix

**Comment by astefanutti** [comment]
@jwforres I've just created hawtio/openshift-jvm#4 that should correct the path and add a couple of other missing resources. Once @gashcrumb feedback is OK, I presume he'll do a release, as v1.0.45 contains a number of fixes and improvements compared to v1.0.42.

**Pull request merged**